### ISSUE TITLE
Is there another source to get the users I follow than 'recent feed'?

### ISSUE THREAD

**Issue by manguszta** [instabot-py/instabot.py#779]
In the past 2 days, I tried to improve the bot (I'm new to Phyton, but familiar to PHP, so I learn fast).

My goal was to
- unfollow those who don't follow me back
- use the bot in the original mode
- make the bot unfollow more users than it follows (so I don't have to use an app to do this work)

I have pretty much successed, BUT
- Unfortunately I couldn't figure out if there is another source to get the users I follow than **media_on_feed**...
- I understood that the bot gets the users I follow from here, and it's called the recent feed: https://www.instagram.com/?__a=1
- I can see there are limited number of users here (about 60 if I search the word "username").

To get a user unfollowed (who doesn't follow me back) I wrote a loop, it doesn't just check if a user follows me back, it checks **UNTIL** it finds a user who doesn't follow me, and then unfollows it. 
I used **unfollow_whitelist** to register the users who actually follow me, so they are checked only once (I add to unfollow_whitelist even the ones I just followed with the bot, otherwise the bot would unfollow it immediately).
What I suspect is 'get_media_id_recent_feed' function picks only up to 12 (of 60) users at a time.
The result is my loop just breaks after the 12th unsuccessed unfollow, because it can't find more users to unfollow. The bot stops as well (I didn't have time to figure out how to avoid this).

I always have at least 900 to 2000 users who don't follow me back. My obvious question would be the one I put in the issue's name: is there another source to get the users **I follow** than my recent feed?
Is there a list of IDs and names (that my account follows) that I could search in?

**Comment by manguszta** [comment]
It seems unfollowing method was too fast for the Instagram feed refresh, and unfollow_whitelist grew too large. It simply stops the bot when there is no one left to unfollow. I implemented a 4-second timer to the loop to avoid the bot from stopping for now. But I'd really like to browse my follower list, not my Instagram feed.

**Comment by disprozzy** [comment]
I use https://github.com/LevPasha/Instagram-API-python to get list of "Following" and than use instabot to unfollow
not sure if it's possible to get the list using instabot itself

**Comment by manguszta** [comment]
@disprozzy well, I installed, but there is no InstagramAPI.py that user_followings.py would require. no test.py either..

**Comment by manguszta** [comment]
Okay, I found a copy on the internet. I managed to set up the 'following list' of users, now trying to implement it in instabot.

**Comment by manguszta** [comment]
It seems I implemented InstagramAPI.py to instabot successfully.

What my bot does:
- Adds newly followed users to unfollow_whitelist 
- Gets user followings list through the API, saves it into a global list of names and IDs
- Instead of randomly choosing a user from recent feed, it is choosing a user from the user following list
- Compares user's ID with the unfollow_whitelist, skips user if 
- Searches the user's other data by the user's id (e.g. is following back)
- If the user is a follower of my account, it adds the user to the unfollow_whitelist, so the bot won't check that user again
- Immediately checks again for another user if the unfollowing was unsuccessful, but only until 5 seconds, so the bot won't stuck in a loop and the running can continue with other stuff (liking, following)

I'm going to test it as I can see there are some successful finds on users who don't follow back, still the bot (for some reason) skips their unfollowing. Strange, but on the other hand, it works nicely.

**Comment by manguszta** [comment]
The unsuccessful unfollowing is because this is not true: "follow.status_code == 200"
So the server doesn't respond with OK on unfollows.

**Comment by manguszta** [comment]
Interesting thing happens, when only a few people are left in the **followings list**.
I wrote another call to the script for repopulating the followings list after the len(followings_list) has gone to zero. But for some reason there are 3 accounts in the list that the bot simply just can't unfollow, because "Except on get_info!". Because I wrote a 5 second loop for unfollows, it randomly chooses between the 3 remaining accounts and tries to unfollow, only until 5 seconds. I think the only way to solve this problem is to write the usernames to exception message, and manually unfollow them from the app.
![kepkivagas](https://user-images.githubusercontent.com/18244325/31796427-be2aa716-b529-11e7-90e7-0ee73df4891f.PNG)

**Comment by finestshops** [comment]
maybe those accounts are private and did not approve you

**Comment by disprozzy** [comment]
maybe those accounts are private and they did not approve you?

**Comment by HenryCooperGithub** [comment]
**Please update to the most recent version of instabot.py**

If your problem still persists open a new issue and attach the following information:

- Your bot configuration (parameters)
- Full console output
